src/main/java/com/google/devtools/build/lib/syntax/Identifier.java

```diff
@@ -117,4 +117,9 @@ EvalException createInvalidIdentifierException(Set<String> symbols) {
     String suggestion = SpellChecker.didYouMean(name, symbols);
     return new EvalException(getLocation(), "name '" + name + "' is not defined" + suggestion);
   }
+
+  /** @return The {@link Identifier} of the provided name. */
+  public static Identifier of(String name) {
```

Comment:
Now that it no longer canonicalizes the string, this method is no longer necessary, nor all the updates to the call sites.

Reply:
even though it's not technically necessary I believe that it makes the code look more fluent (`Identifier.of('some_variable')` which reads like English vs `new Identifier('some_variable')`) and it's also a common practice for important library code, which it makes code evolution much easier as it makes it easier to maintain open/closed design principle - for instance if one of the use cases can be handled more efficiently by having a specialized `Identifier` (like the one for `Star` or `StarStart` for example), it's possible to return that specialized instance from the factory method, but if constructor is used, all the ifs would have to go inside of the class itself. Also, in case things like interning do end up being useful, it would be easy to do this in a factory method, whereas constructor always returns a new instance of the object.

In other words, I see only advantages of using factory method and no downsides (JIT will inline it anyways, so no perf overhead). Do you happen to have some concerns about usage of a factory method?

Reply:
Reading like English is not a goal. A programmer should be as fluent in their main programming language as in their mother tongue. The meaning of `new T(x)` needs no explanation, whereas `T.of(x)` is another function one must read and remember, even if it is just one drop of needless abstraction in an existing ocean. I read many wondrous claims about the JVM making the costs of abstraction disappear, but few supported by measurement. Blaze spends 2 seconds of CPU before control reaches main, largely due to code size.

I'll defer to the Bazel team for style issues though.

Reply:
I think this makes little difference. I'm fine with either keeping or reverting it.